### PR TITLE
[WIP] first port of simple BV exporter from philistine

### PR DESCRIPTION
**Pull request by choldgraf**
This PR extracts the BV writing code from philistine, and removes the dependence (except for testing) on MNE-python. It's a WIP but should give an idea of where this is heading!

### CODE REVIEW

File: pybv/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018 The BIDS Community
+# License: BSD (3-clause)
+"""pybv: a lightweight BrainVision I/O tool"""
+
+__version__ = '0.1'
```
Comment (sappelhoff) [inline pybv/__init__.py]: I would suggest folowing https://semver.org/ but starting at 0.0.0 for now to make it really obvious that this is an alpha stage.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                       'License :: OSI Approved',
+                       'Programming Language :: Python',
+                       'Topic :: Software Development',
+                       'Topic :: Scientific/Engineering',
+                       'Operating System :: Microsoft :: Windows',
+                       'Operating System :: POSIX',
+                       'Operating System :: Unix',
+                       'Operating System :: MacOS'],
+          platforms='any',
+          packages=package_tree('pybv'),
```
Comment (sappelhoff) [inline setup.py]: why not use `setuptools.find_packages` here?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+DESCRIPTION = descr
+MAINTAINER = 'The BIDS Community'
+MAINTAINER_EMAIL = 'TODO'
+URL = 'https://github.com/bids-standard/pybv'
+LICENSE = 'BSD (3-clause)'
+DOWNLOAD_URL = 'https://github.com/bids-standard/pybv'
+VERSION = version
+
+
+def package_tree(pkgroot):
```
Comment (sappelhoff) [inline setup.py]: can be replaced by `setuptools.find_packages`

File: docs/Makefile [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
+SPHINXBUILD   = sphinx-build
+SPHINXPROJ    = philistine
```
Comment (sappelhoff) [inline docs/Makefile]: philistine?

File: docs/conf.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+# Add any paths that contain custom static files (such as style sheets) here,
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+html_static_path = ['_static']
+
+
+# -- Options for HTMLHelp output ------------------------------------------
+
+# Output file base name for HTML help builder.
+htmlhelp_basename = 'philistinedoc'
```
Comment (sappelhoff) [inline docs/conf.py]: philistinedoc?

File: docs/conf.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    #
+    # 'figure_align': 'htbp',
+}
+
+# Grouping the document tree into LaTeX files. List of tuples
+# (source start file, target name, title,
+#  author, documentclass [howto, manual, or own class]).
+latex_documents = [
+    (master_doc, 'philistine.tex', 'philistine Documentation',
+     'Phillip M. Alday', 'manual'),
```
Comment (sappelhoff) [inline docs/conf.py]: needs revising

File: docs/conf.py [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+# -- Options for Texinfo output -------------------------------------------
+
+# Grouping the document tree into Texinfo files. List of tuples
+# (source start file, target name, title, author,
+#  dir menu entry, description, category)
+texinfo_documents = [
+    (master_doc, 'pybv', 'pybv Documentation',
+     author, 'pybv', 'One line description of project.',
+     'Miscellaneous'),
+]
```
Comment (sappelhoff) [inline docs/conf.py]: probably needs revsising?

File: docs/index.rst [hunk 8]
```diff
@@ -0,0 +1,18 @@
+.. philistine documentation master file, created by
```
Comment (sappelhoff) [inline docs/index.rst]: philistine? needs revising

**Comment by choldgraf** [comment]
@sappelhoff thanks for the input! Can you please hold off on making any changes to this branch until it is no longer WIP or you run it by me first? I've been making more changes to the code over night, and didn't realize that you'd pushed, so there might be merge conflicts now :-/  I was just showing the code so people could comment on the general approach, it's definitely still not finished

**Comment by choldgraf** [comment]
re: docs etc, yep I totally agree there, just haven't gotten around to changing the docs from the philistine copypasta yet :-)

**Comment by sappelhoff** [comment]
> Can you please hold off on making any changes to this branch until it is no longer WIP or you run it by me first?

yea, sorry 😬

File: setup.py (hunk 2 above)
Comment (choldgraf) [reply setup.py]: For this kinda stuff that's just copypastad over, can we iterate on another PR? I wanna get this PR out the door and set up with testing, and then we can start iterating bit-by-bit from there. That OK?

Comment (sappelhoff) [reply setup.py]: +1

**Comment by choldgraf** [comment]
OK, got the tests passing and updated the docs with a simple example. Note that MNE recently changed how they want people to handle events and annotations, so I'm using the new "annotations" interface in the testing etc.

@sappelhoff if you wanna give it a shot and/or make edits to the branch, go for it now! Alternatively, I'm also happy to merge this and then start iterating in subsequent PRs once people are happy with this current implementation.

**Comment by choldgraf** [comment]
docs are here: https://17-157434681-gh.circle-artifacts.com/0/html/index.html

(it's only one page)

**Comment by choldgraf** [comment]
@agramfort can you look at the built docs and let me know if that's how annotations/events are meant to be handled in MNE? It took a bit of fiddling to figure out why I started getting all these deprecation warnings :-)

**Comment by sappelhoff** [comment]
@choldgraf see https://github.com/choldgraf/pybv/pull/1

after integration of these changes and green CI, +1 for merge from my side.

Good work there, that was quick! :-)

**Comment by choldgraf** [comment]
LGTM...shall we give it a merge and iterate from there?

**Comment by teonbrooks** [comment]
this exporter seems to only depend on `mne` for the testing but not the core code. is there another bv dataset publicly accessible maybe on osf?

File: README.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+**See the [pybv documentation](https://pybv.readthedocs.io) for more
+information.**
+
+**ALPHA SOFTWARE**. This package is currently in its early stages of iteration.
+It may change both its internals or its user-facing API in the near future. Any
+feedback and ideas on how to improve either of these is more than welcome!
+
+## Acknowledgements
+
+This package was originally adapted from [palday](https://github.com/palday)'s
```
Comment (teonbrooks) [inline README.md]: https://gitlab.com/palday

**Comment by choldgraf** [comment]
It's not actually using mne for the dataset, just to double check the round-trip conversion to brain vision and back in to mne... I feel like it's reasonable to keep mne as a "first class citizen" for the sake of testing since it's so common in python, no? Wdyt?

**Comment by teonbrooks** [comment]
I was just reading within the docs that it did not have dependency on mne but I saw it included in `requirements.txt`. i was just commenting on the consistency.

I mean, you know I am a major fanboy of mne. I didn't know if this was supposed to be interoperable with mne but independent from it.

**Comment by teonbrooks** [comment]
also, maybe I just don't know the function of the `requirements.txt` 😅 just saw that `mne` is not part of the `setup.py`

File: README.md (hunk 9 above)
Comment (sappelhoff) [reply README.md]:
```suggestion
This package was originally adapted from [palday](https://gitlab.com/palday)'s
```

Comment (sappelhoff) [reply README.md]: I think it's probably "https://palday.bitbucket.io/" if we want to be very strict :-D

**Comment by sappelhoff** [comment]
> I didn't know if this was supposed to be interoperable with mne but independent from it.

Definitely independent and as few dependencies as possible (in my opinion)


>  maybe I just don't know the function of the requirements.txt

I think it's mainly there for CircleCI's automated testing?

> I feel like it's reasonable to keep mne as a "first class citizen" for the sake of testing since it's so common in python

I think for this "alpha" release it's fine - but IMO we should get rid of MNE as soon as we have an own reader :-)

> is there another bv dataset publicly accessible maybe on osf?

yes! There is one even with a CC0 license! :-) see #4

**Comment by sappelhoff** [comment]
I added Teon's point (palday website) in https://github.com/choldgraf/pybv/pull/2

and also wrote down some installation instructions.

**Comment by choldgraf** [comment]
+1 on removing the MNE dependency once we have a reader...though depending on how complex reading the data will be, it may be worth coordinating with MNE on a reader so that we don't duplicate a lot of complex code...

**Comment by sappelhoff** [comment]
> may be worth coordinating with MNE on a reader so that we don't duplicate a lot of complex code

+1

I'll merge this one for now. Thanks @choldgraf !

**Pull request merged**